packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -78,8 +80,16 @@ function outerHTMLGetterPatched(this: Element) {
     return getOuterHTML(this);
 }
 
-function attachShadowPatched(this: Element, options: ShadowRootInit): SyntheticShadowRootInterface {
-    return attachShadow(this, options);
+function attachShadowPatched(
+    this: Element,
+    options: ShadowRootInit
+): SyntheticShadowRootInterface | ShadowRoot {
```

Comment:
```suggestion
): ShadowRoot {
```

I would also recommend replacing `SyntheticShadowRootInterface` by an ambient type. But it's something for another PR.